### PR TITLE
feat: replaces console.warn instead throwing error when event is not valid

### PR DESCRIPTION
**Pull request by vanGalilea**
## CLA
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] Code changes are tested

## Description of the changes, What, Why and How?
Motive of this fix is the following issue https://github.com/GetStream/stream-chat-js/issues/869

## Changelog
- replaces console.warn instead throwing an error when event is not valid

### CODE REVIEW

File: src/channel.ts [hunk 1]
```diff
@@ -1119,7 +1119,7 @@ export class Channel<StreamChatGenerics extends ExtendableGenerics = DefaultGene
     const key = callbackOrNothing ? (callbackOrString as string) : 'all';
     const valid = isValidEventType(key);
     if (!valid) {
-      throw Error(`Invalid event type ${key}`);
+      console.warn(`Invalid event type ${key}`);
```
Comment (ferhatelmas) [inline src/channel.ts:1122]: early return is missing

Comment (vishalnarkhede) [reply src/channel.ts:1122]: No, we want to continue executing listeners here. The reason being this `throw error` creates blocker for custom events.

Comment (ferhatelmas) [reply src/channel.ts:1122]: It's not executing code path but changing listeners so it will put an invalid key

Comment (vishalnarkhede) [reply src/channel.ts:1122]: I am wondering if we should simply remove this console.warn as well.

Comment (ferhatelmas) [reply src/channel.ts:1122]: so do you want to put whatever given by user? makes sense to drop then

Comment (vishalnarkhede) [reply src/channel.ts:1122]: Yeah!! cc @vanGalilea

Comment (vanGalilea) [reply src/channel.ts]: Ready guys!

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**